### DISCUSSION TITLE
Server owners can ban annoying clients

### DISCUSSION THREAD

**Discussion by A5005** [jamulussoftware/jamulus#1578] [Ideas]
This suggestion could polarize, please say what you think about it.

**Problem**
Everyone knows it: You have a great session on a public server and suddenly annoying clients come in and do a soundcheck parallel to the song or intentionally disrupting the session.

**Describe the solution you'd like**
As a server owner, I would like to have the opportunity to ban such clients.

**Comment by gene96817** [comment]
For simplicity, there are two kinds of annoying visitors: 1) naive or clueless and 2) malicious.
I'd like to believe the first case can be solved with education. In the second case, the malicious can't be stopped until we have a technically sound (serious?) authentication.  This is one scenario that is addressed in the jam session management discussion.

**Comment by npostavs** [comment]
I wrote a little script to block people by IP in the firewall (Linux specific). But I haven't had any occasion to use it so far. Sometimes people with misconfigured audio setups come and make noise, but it's easy enough to mute them (or even set new client level to 0 so new people start out muted). And anyway, those people tend to disconnect quickly once they realize something is wrong.

```sh
#!/bin/sh

if [ -z "$1" ] ; then
    echo "Usage: $0 --reset|<ip-address[:port]>" >&2
    exit 1
fi

case "$1" in
    (--reset) /etc/nftables.conf ;;
    (*:*) nft insert rule inet filter input ip saddr "${1%:*}" udp sport "${1#*:}" reject ;;
    (*) nft insert rule inet filter input ip saddr "$1" reject ;;
esac
```

**Reply by A5005** [reply]
How can you determine the IP of the connected (annoying) client?
The problem with "set new client level to 0" is that it also fades out constructive musicians.

**Reply by npostavs** [reply]
Oh right, I added the IP info to the html status file, see "Add IP info to HTML channel list" in https://github.com/jamulussoftware/jamulus/compare/master...npostavs:np.server (not linking commit directly because I rebase that branch every so often). Then I look at the status file with `cat` (since I'm not running a web server; if you are serving your status file, this may be a bad idea since you would end up broadcasting client IPs to the whole web). If the server is in GUI mode, you can already see this info.

But anyway, like I said, I haven't ever had occasion to use this. Annoying clients never stayed long enough.

```diff
modified   src/server.cpp
@@ -1749,9 +1749,12 @@ void CServer::WriteHTMLChannelList()
             // write entry for each connected client
             for ( int i = 0; i < iMaxNumChannels; i++ )
             {
-                if ( vecChannels[i].IsConnected() )
+                CChannel& channel = vecChannels[i];
+                if ( channel.IsConnected() )
                 {
-                    streamFileOut << "  <li>" << vecChannels[i].GetName().toHtmlEscaped() << "</li>\n";
+                    streamFileOut << "  <li>" << channel.GetName().toHtmlEscaped()
+                                  << " (" << channel.GetAddress().toString() <<")"
+                                  << "</li>\n";
                 }
             }
 ```
